Formula/icu4c.rb

```diff
@@ -8,8 +8,10 @@ class Icu4c < Formula
 
   livecheck do
     url :stable
-    strategy :github_latest
     regex(%r{href=.*?/tag/release[._-]v?(\d+(?:[.-]\d+)+)["' >]}i)
+    strategy :github_latest do |page, regex|
+      page.scan(regex).map { |match| match&.first&.gsub(/-/, ".") }
+    end
```

Comment:
```suggestion
    strategy :git do |tags, regex|
      tags.map { |tag| tag[regex, 1]&.gsub("-", ".") }.compact
    end
```

From what I'm seeing, checking the Git tags seems fine and I don't think it's necessary to check the "latest" release on GitHub (we only use `GithubLatest` when it's available, correct, and necessary). Tags like `release-68-2` are stable and unstable releases have a suffix (e.g., `release-68-alpha`, `release-68-rc`, `release-66-preview`, etc.), so only stable releases will be matched if we update the regex like so: `regex(/^release[._-]v?(\d+(?:[.-]\d+)+)$/i)`

Let's switch to using the Git tags for now, as we can always return to using `GithubLatest` if we run into any problems.